### PR TITLE
Fix English errors in docs

### PR DESCRIPTION
**Pull request by mks-h**
The documentation has a bunch of grammatical errors, so I am fixing the obvious ones. The text is also worded quite awkwardly, so I may also rephrase it in some places.

### CODE REVIEW

**Comment by mks-h** [comment]
The CI caught none of the real spelling errors, yet it blocks me on using "lifecycle" instead of "life cycle" even tho in the technical documentation filed it is a much more popular variant. I'd advise against having such a CI.

**Comment by NigelBreslaw** [comment]
Thank you for this! I can see cSpell is misconfigured which is letting some of the typos you found go undetected. I've created a separate PR to fix this and a few other typos it found https://github.com/slint-ui/slint/pull/7233

Once this goes through there will be some conflicts with this PR to fix. Regarding 'life cycle' we are following the main US dictionaries such as Merriam Webster and it uses life cycle as separate words. I did some quick googling and it is the most common usage of the term so we will stick with that. Let's keep life cycle and then get the rest of your fixes in. Thanks again for spotting these issues.

**Comment by NigelBreslaw** [comment]
Okay the cSpell fix is in. If you are okay to fix the conflicts we can then get this PR merged in?

**Comment by NigelBreslaw** [comment]
It's still failing on a couple of typos src/content/docs/guide/language/concepts/slint-language.mdx:88:58 - Unknown word (declartive)
src/content/docs/guide/language/concepts/slint-language.mdx:89:74 - Unknown word (behaviour)

You can run the spell check locally by `cd docs/astro && pnpm i && pnpm spellcheck` presuming you have node and pnpm installed

File: docs/astro/src/content/docs/guide/language/concepts/slint-language.mdx [hunk 1]
```diff
@@ -48,16 +48,16 @@ document.body.appendChild(buttonWithListener);
...
+It's quite abstract. For example once a few more buttons and components are added its almost impossible
 to be able to think how the real interface would look. It's hard to think about the design of a UI
 using this kind of code.
 
 It's also too complex to edit without understanding how to code. This means UI designers cannot work
 hands on to ensure all their design intent is implemented. They are forced to use other tools and frameworks
-to crate prototypes and design guides that may look or behave differently to the software used to implement
-the production UI. It should not have to be this way.
+to create prototypes and design guides that may look or behave differently to the software used to implement
+the production UI. It doesn't have to be this way.
```
Comment (mks-h) [inline docs/astro/src/content/docs/guide/language/concepts/slint-language.mdx:60]: `the software used to implement the production UI` is a bit awkward. Can I change it to `the actual UI implementation`?

Comment (NigelBreslaw) [reply docs/astro/src/content/docs/guide/language/concepts/slint-language.mdx:60]: Please do change that

**Comment by mks-h** [comment]
There are still some issues, but obvious mistakes are fixed.

**Comment by NigelBreslaw** [comment]
This is great for now. Once CI passes I will merge it in. Thanks again for both the better text and showing the spell check was not setup correctly.

**Pull request merged**